CITATION.cff

```diff
@@ -27,4 +27,4 @@ authors:
 version: 0.4.1
 date-released: "2022-04-09"
 license: Apache-2.0
-repository-code: "https://github.com/metaopt/TorchOpt"
+repository-code: "https://github.com/metaopt/torchopt"
```

Comment:
Not the github repo url

Reply:
We'd better use lower cases here, as well as the PyPI package. That's a convention.

I'm fine with both CamelCases or lowercases, I can revert the changes. But I prefer the lowercases.

GitHub will automatically redirect the URLs. Now we have the CamelCased repo URL, and <https://github.com/metaopt/torchopt> will redirect to TorchOpt.